convert_client.c

```diff
@@ -303,8 +304,7 @@ _read_convert(socket_state_t *state, long *result, bool peek, int fail_errno)
 	}
 
 	if (length) {
-		uint8_t			buffer[length + offset];
-		struct convert_opts	opts;
```

Comment:
`struct convert_opts` could also be bigger to host `Convert Extended TCP Header` to avoid all these malloc/free for each connection? But certainly fine to do these malloc/free

Reply:
Indeed, this could be an alternative. As there are multiple variable-size TLVs in Convert, in addition to the Extender TCP HDR TLV, we would have to use bigger structures for each type, thus using much more space than needed. And for some TLVs, especially the Error TLV, it is hard to tell what the maximum size is. For instance, the Malformed Message (1) error code must echo the entire Convert message. For this reason, I felt it was cleaner to dynamically allocate to the appropriate size and kindly ask the caller to free it.

Reply:
OK, I thought it was only for the connect but indeed, in theory we could even send 1500B there.

If it was for just 10 bytes, it would have certainly be easier not to be dynamic but here it is different. We could also use other things than malloc but that's a detail for now!